### PR TITLE
Added update wechat user info endpoint

### PR DESCRIPTION
**Pull request by XeniaLu**

### CODE REVIEW

File: src/gapp_login/wechat/endpoints.py [hunk 1]
```diff
@@ -83,6 +96,42 @@ async def login_wechat(
...
+    iv: str = Form(...),
+    app_id: str = Form(...),
+):
+    encrypted_data = base64.b64decode(encrypted_data)
+    iv = base64.b64decode(iv)
+    identity = await (
+        WeChatIdentity.query.select_from(User.outerjoin(WeChatIdentity))
+        .where(User.id == user.id)
+        .where(WeChatIdentity.wechat_app_id == app_id)
+        .where(db.func.starts_with(WeChatIdentity.idp, "WECHAT"))
```
Comment (XeniaLu) [inline src/gapp_login/wechat/endpoints.py]: @fantix Is there any way to get correct identity without pass the app_id paramter.

Comment (fantix) [reply src/gapp_login/wechat/endpoints.py]: I think we could add `client_id` to the authorization code. I'll add to authlib-gino with a PR.

Comment (XeniaLu) [reply src/gapp_login/wechat/endpoints.py]: Okay. Then I will update this PR after that.

Comment (XeniaLu) [reply src/gapp_login/wechat/endpoints.py]: fixed

File: src/gapp_login/wechat/endpoints.py [hunk 2]
```diff
@@ -44,23 +53,23 @@ async def login_wechat(
...
     async with db.transaction() as tx:
-        result = await (
+        user = await (
             WeChatIdentity.outerjoin(User)
             .select()
             .with_for_update(of=WeChatIdentity)
             .where(WeChatIdentity.sub == openid)
             .where(WeChatIdentity.idp == idp)
-            .gino.load((WeChatIdentity, User))
+            .gino.load((User.load(current_identity=WeChatIdentity)))
```
Comment (fantix) [inline src/gapp_login/wechat/endpoints.py]: (trivial) redundant parenthesis

File: src/gapp_login/wechat/endpoints.py [hunk 3]
```diff
@@ -86,12 +99,49 @@ async def login_wechat(
...
+        raise HTTPException(status.HTTP_400_BAD_REQUEST, "Bad wechat AppID")
+
+    if open_id != identity.sub:
+        raise HTTPException(status.HTTP_400_BAD_REQUEST, "Wechat openId mismatched")
+
+    await identity.update(wechat_user_info=data, wechat_unionid=unionid,).apply()
+    return dict(data=data)
+
+
+@router.get("/users/wechat")
```
Comment (fantix) [inline src/gapp_login/wechat/endpoints.py:133]: (future) authlib-gino 0.4 would probably want to include this in the user_info endpoint, but we could use this as for now

**Review by fantix — APPROVED** [review]

File: src/gapp_login/wechat/endpoints.py (hunk 2 above)
Comment (XeniaLu) [reply src/gapp_login/wechat/endpoints.py]: fixed

File: src/gapp_login/wechat/endpoints.py (hunk 3 above)
Comment (XeniaLu) [reply src/gapp_login/wechat/endpoints.py:133]: Ok~

**Pull request merged**